planner/mapper/mapper.go

```diff
@@ -801,7 +801,7 @@ func resolveInnerFilterDependencies(
 			enumerable.New(existingFields),
 		)
 
-		matchingFields := enumerable.Where(allFields, func(existingField Requestable) (bool, error) {
+		matchingFields := enumerable.Where[Requestable](allFields, func(existingField Requestable) (bool, error) {
```

Comment:
This one annoyed me quite a bit.  It is a limitation of the Go compiler to infer the type from an interface with a Generic. I hope they fix this in 1.20 or something as there should be no reason to have to specify this explicitly here.

Reply:
Hmm, can you double check this? I'm fairly certain it can infer the type, since the `allFields` is of type `Enumerable[Requestable]` and the func takes a `Requestable` both of which the compiler will use to infer the type. I just checked it [here](https://go.dev/play/p/XcLpTOf-y5o) with a stripped down example, and it inters correctly.

Reply:
Compiler still says no.  Your playground example is incomplete as you have only one interface, this happens when extending an interface.

Reply:
Ahhhhh, I see now, forgot that `Concat` returns a diff type than `Enumerable[Requestable]`